### PR TITLE
Add support for definition refs in algorithm calls

### PR DESCRIPTION
**Pull request by judofyr**
This would be a great additional feature to the new definition support!

### CODE REVIEW

**Comment by judofyr** [comment]
Ah, I see now that there's a bunch of test failures. Consider this more of a feature request. Maybe you know of a better way to structure the grammar to implement this.

**Comment by leebyron** [comment]
Most of our test cases are integration tests - so new behavior changes need to record those tests. If you run `npm run record-test` you'll get updated test files to include with your PR

**Comment by judofyr** [comment]
Good to know, but I'm not quite sure if I intended these changes. It turned a lot of `Variable` nodes into `Terminal` which doesn't sound good.

**Comment by leebyron** [comment]
Though I'm curious to know the use case, which might help me review the grammar. The original intent was for call arguments to be considered as locally scoped variables, and this extends beyond that.

File: src/grammar.pegjs [hunk 1]
```diff
@@ -504,7 +504,7 @@ htmlTag = tag:$('<' '/'? [a-z]+ [^>]* '>') {
   });
 }
 
-reference = '{' !('++'/'--') _ ref:(call / value / token)? _ close:'}'? {
+reference = '{' !('++'/'--') _ ref:(call / token / value)? _ close:'}'? {
```
Comment (leebyron) [inline src/grammar.pegjs:507]: This flip is probably what caused the unexpected changes you're seeing in tests. PEG try to parse in order and there's some overlap between these two rules

Comment (judofyr) [reply src/grammar.pegjs:507]: Yep. And the reason for flipping it was because it started parsing the closing parenthesis as part of the token.

**Comment by judofyr** [comment]
> Though I'm curious to know the use case, which might help me review the grammar. The original intent was for call arguments to be considered as locally scoped variables, and this extends beyond that.

Constants/enums! I want to be able to define a constant/enum and refer to it.

**Pull request closed without merging**